client/my-sites/plugins/related-plugins/index.tsx

```diff
@@ -0,0 +1,101 @@
+import { Gridicon, Button } from '@automattic/components';
+import { useTranslate } from 'i18n-calypso';
+import { useMemo } from 'react';
+import { RelatedPlugin } from 'calypso/data/marketplace/types';
+import { useGetRelatedPlugins } from 'calypso/data/marketplace/use-get-related-plugins';
+import { IntervalLength } from 'calypso/my-sites/marketplace/components/billing-interval-switcher/constants';
+import PluginIcon from 'calypso/my-sites/plugins/plugin-icon/plugin-icon';
+import { PluginPrice } from 'calypso/my-sites/plugins/plugin-price';
+import PreinstalledPremiumPluginPriceDisplay from 'calypso/my-sites/plugins/plugin-price/preinstalled-premium-plugin-price-display';
+import { useSelector } from 'calypso/state';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+import './style.scss';
+import { getFirstCategoryFromTags } from '../categories/use-categories';
+
+type RelatedPluginProps = {
+	slug: string;
+	size: number;
+	seeAllLink?: string;
+	options: {
+		enabled?: boolean;
+		staleTime?: number;
+		refetchOnMount?: boolean;
+	};
+};
+
+function RelatedPlugins( { slug, size = 4, seeAllLink, options }: RelatedPluginProps ) {
+	const translate = useTranslate();
+	const { data: relatedPlugins } = useGetRelatedPlugins( slug, size, { ...options } );
+
+	return (
+		<div className="related-plugins">
+			<div className="related-plugins__header">
+				<h2>{ translate( 'Related plugins' ) }</h2>
+				{ seeAllLink && (
+					<Button borderless primary>
+						<span>{ translate( 'See all' ) }</span>
+						<Gridicon icon="chevron-right" />
+					</Button>
+				) }
+			</div>
+			<div className="related-plugins__list">
+				{ relatedPlugins &&
+					relatedPlugins.map( ( plugin: RelatedPlugin ) => (
+						<RelatedPluginCard key={ plugin.slug } plugin={ plugin } />
+					) ) }
+			</div>
+		</div>
+	);
```

Comment:
> Is the `seeAllLink` used here as a boolean just stating if we want to show or not the `See All` link? If that's the case we will want to change the signature `seeAllLink?: string;`.
I thought about this a bit but couldn't find a better type for what we'd want here, could you help with what would be a sound signature in this case? I think the `string | undefined` signature is sufficient here because if there's no `seeAllLink` prop passed, it'll be set to `undefined`, and the condition won't be triggered. Otherwise, if a URL is passed, then it'll be rendered.

> If this actually will contain the string with the link, which I think is the case, we want to point this button to the URL, right? I think the `onClick` callback is missing here.
Yes, good point, I've added this in a49d1daeb1021e622ae6ff7fb0b426a64c46a0b9.